feat: make Booker Atom compatible with dynamic bookings [platform]

hackice20 opened the pull request:
## What does this PR do?

- Fixes #25295  (GitHub issue number)

**Key Changes:**
- Added `entity` prop support to Booker Atom with full type definitions
- Implemented transparent `orgSlug` resolution that falls back to event data when not explicitly provided
- Added comprehensive documentation for dynamic bookings and entity configuration
- Updated BookerPlatformWrapper to handle entity context for availability resolution

## Visual Demo (For contributors especially)
<img width="1920" height="1080" alt="pr2" src="https://github.com/user-attachments/assets/eb9ad2ba-10d4-42aa-ba80-c0acd660ccea" />
<img width="1920" height="1080" alt="pr1" src="https://github.com/user-attachments/assets/0055d864-b2fa-4b52-8262-f2c2c9303e07" />

GET /api/trpc/slots/getSchedule

```
{
  "json": {
    "isTeamEvent": false,
    "usernameList": ["pro", "pro2"],
    "eventTypeSlug": "tennis-class",
    "startTime": "2025-10-31T18:30:00.000Z",
    "endTime": "2025-12-31T18:29:59.999Z",
    "timeZone": "Asia/Kolkata",
    "duration": "30",
    "rescheduleUid": null,
    "orgSlug": null,
    "teamMemberEmail": null,
    "routedTeamMemberIds": null,
    "skipContactOwner": false,
    "routingFormResponseId": null,
    "email": null,
    "embedConnectVersion": "0",
    "_isDryRun": false
  },
  "meta": {
    "values": {
      "orgSlug": ["undefined"],
      "teamMemberEmail": ["undefined"],
      "routingFormResponseId": ["undefined"]
    }
  }
}
```
As you see above the `usernameList` and `orgSlug` . the orgSlug will be set to exact orgName when group is under Org.

## Mandatory Tasks (DO NOT REMOVE)

- [X] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [X] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## Checklist

- [X] I have read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- [X] My code does follow the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have checked if my changes generate no new warnings

hackice20 committed:
Add entity configuration support for booker atom

supalarry commented on docs/platform/atoms/booker.mdx:
```diff
@@ -24,6 +24,66 @@ export default function Booker( props : BookerProps ) {
+
+<Note>
+**Requirements for Dynamic Bookings:**
+- All specified users must have `allowDynamicBooking` enabled in their settings
```
This is good information to keep in documentation . However, we need to mention that it is enabled by default as true so it's something to be aware only if dynamic bookings are not working.

supalarry commented on docs/platform/atoms/booker.mdx:
```diff
@@ -24,6 +24,66 @@ export default function Booker( props : BookerProps ) {
+
+### Dynamic Booking with Organization Context
+
+When booking within an organization, provide the `entity` prop to ensure correct availability resolution:
```
Since you added function `resolveOrgSlug` which resolves it from the `event.data` we don't need to pass entity anymore.

supalarry reviewed (changes requested):
Thank you for your PR great work. I really like the `function resolveOrgSlug` and would like to ask - can you, please, narrow this PR down? We really just need:
1. `function resolveOrgSlug` in the `BookerPlatformWrapper.tsx` that resolves org slug.
2. The docs just need to mention what are dynamic events, when are they useful, that `allowDynamicBooking` has to be true for each user but its true by default, and that to use them one must simply pass username as array containing usernames.
3. We don't need to pass entity since its resolved automatically, so can you remove org slug resolve via entity and just rely on event.data ? So that means we don't need public facing docs on entity.

I tested it locally and it runs fine without passing event slug or entity
<img width="1001" height="487" alt="Screenshot 2025-12-05 at 12 13 02" src="https://github.com/user-attachments/assets/10c68c9a-b191-469c-8a09-364e14c78021" />

Once again, thanks for the contribution and let me know when can re-review.

hackice20 committed:
Merge remote-tracking branch 'origin/main' into feat/booker-atom-dynamic

hackice20 committed:
Simplify dynamic booking support: auto-resolve orgSlug from event data

- Simplified resolveOrgSlug to only use event.data (removed entityFromProps)
- Removed public entity prop documentation (resolved automatically)
- Updated docs to mention allowDynamicBooking is enabled by default
- Removed BookerEntityConfig from public exports

hackice20 committed:
Merge branch 'main' into feat/booker-atom-dynamic

supalarry commented on packages/platform/atoms/booker/BookerPlatformWrapper.tsx:
```diff
@@ -198,7 +217,7 @@ const BookerPlatformWrapperComponent = (
     bookingUid: props.bookingUid ?? null,
     layout: layout,
-    org: props.entity?.orgSlug,
+    org: resolvedOrgSlug,
```
Ah, my bad - i forgot that we already pass `props.entity?.orgSlug` to org. Which means that you were right - we need `resolveOrgSlug` to first check if `props.entity?.orgSlug1 is defined and return that and if thats not returned return from event.data.

Sorry for the confusion. We do this to ensure that code continues to work the same for people who are already passing `props.entity?.orgSlug` but we dont document it so that new users rely on the automatically inferred org based on event data.

supalarry commented on packages/platform/atoms/booker/types.ts:
```diff
@@ -39,14 +39,27 @@ export type BookerStoreValues = Omit<
   | "setOrg"
 >;
 
+// Internal entity configuration type - resolved automatically from event data
```
Can we pls remove this comment

supalarry commented on packages/platform/atoms/calendar-view/wrappers/CalendarViewPlatformWrapper.tsx:
```diff
@@ -13,21 +39,115 @@ export type CalendarViewPlatformWrapperAtomPropsForTeam = {
+) => {
+  const isTeamEvent = !!props.teamId;
+  const teamId: number | undefined = props.teamId ? props.teamId : undefined;
+  const username = useMemo(() => {
```
This is a lot of new code added thats basically copy from booker platform wrapper. If this code here needs `orgSlug`, can't we pass org slug from `const setOrg = useBookerStoreContext((state) => state.setOrg)` ? state but just access the set org in state?

supalarry reviewed (changes requested):
Appreciate the fast response : ) Here is my review

hackice20 committed:
Simplify PR per reviewer feedback

hackice20 committed:
Merge branch 'main' into feat/booker-atom-dynamic

hackice20 committed:
Merge branch 'main' into feat/booker-atom-dynamic

supalarry commented on packages/platform/atoms/booker/BookerPlatformWrapper.tsx:
```diff
@@ -304,7 +327,8 @@ const BookerPlatformWrapperComponent = (
       !event?.isPending &&
       event?.data?.id != null,
-    orgSlug: props.entity?.orgSlug ?? undefined,
+    // Use resolved orgSlug for availability - falls back gracefully if not available
```
Can we remove all comments to files added in this PR except documentation? This comment simply restates the code and if the code updates then we also need to maintain this doc, so its easier to delete it.

supalarry reviewed (changes requested):
Review:

hackice20 committed:
Address reviewer feedback: remove duplicate types, clean up comments, update docs

supalarry commented on packages/features/bookings/Booker/types.ts:
```diff
@@ -30,19 +50,8 @@ export interface BookerProps {
-    logoUrl?: string | null;
-    eventTypeId?: number | null;
-  };
+  /** Entity configuration - orgSlug is auto-resolved from event data in Platform Booker */
```
We can remove this comment - this component should not explain what is resolved how because that logic lives upper in hierarchy.

supalarry reviewed (changes requested):
Why are we renaming the exports?

hackice20 committed:
chore: clean booker docs and revert atom exports

hackice20 committed:
Remove comments from BookerProps interfaceO

hackice20 commented:
Hey, @supalarry take a look, let me know what you think
